### DISCUSSION TITLE
Is the NSEC ordering depending on Database ordering?

### DISCUSSION THREAD

**Discussion by klaus-nicat** [PowerDNS/pdns#13758] [Q&A] [answered]
Hi! I am using the Postgresql Backend. There, for NSEC and NSEC3, the `order` column is populated and used to find the respective NSEC(3) records. I wonder, is the selection of the proper NSEC(3) records depending on the SORT BY of the database (and hence depending on the Postgresql `LC_COLLATE`) or is the canonical ordering done internall in PowerDNS?

**Comment by Habbie** [comment, marked as answer]
Neither! For every database, we carefully researched how to make the database to the right thing.

Postgres:

```
CREATE INDEX recordorder ON records (domain_id, ordername text_pattern_ops);
```

and `using ~<~` in the ordername `order by` .

MySQL:

```
  ordername             VARCHAR(255) BINARY DEFAULT NULL,
```

etc.

If you have a custom schema, or are worried, `pdnsutil test-schema` will check a few edge cases that the "usual" collations might get wrong.

**Reply by klaus-nicat** [reply]
Great. Thanks. So I will create our new Postgresql database with LOCALE C.UTF8 to avoid reindexing the next time the en_us LOCALE changes ....
